src/bin/encode.rs

```diff
@@ -257,8 +287,16 @@ fn main_aux() {
         quiet
     };
 
-    for source_path in sources {
-        handle_path(&mut options, source_path, PathBuf::new().as_path());
+    if sources.len() == 0 {
+        let mut buffer = String::new();
+        stdin().read_to_string(&mut buffer)
```

Comment:
So, here, too, why do we block until we have read to the end? Is it just to know the length of the file? If so, I'm not sure we need it all that much.

Reply:
yeah, just to keep the same behavior.  I'll file another issue to remove that part from the progress message.